apps/prairielearn/elements/pl-big-o-input/pl-big-o-input.py

```diff
@@ -119,7 +119,7 @@ def render(element_html: str, data: pl.QuestionData) -> str:
         "constants": constants,
     }
 
-    with open(BIG_O_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
+    with open(BIG_O_INPUT_MUSTACHE_TEMPLATE_NAME, encoding="utf-8") as f:
```

Comment:
What's the reason for this change?

I don't see the rationale for being _less_ explicit here.

Reply:
Not exactly an explanation, but: https://github.com/asottile/pyupgrade?tab=readme-ov-file#redundant-open-modes

Reply:
I'm on the fence about this change. I would be fine disabling this specific rule -- explicit generally seems better. Here is the ruff explanation: https://docs.astral.sh/ruff/rules/redundant-open-modes/

Reply:
FWIW we pretty much never explicitly specify a mode in Node and that's never been a problem for us. I'm fine with this change.